craft_providers/multipass/_ready.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""Multipass Provider Readiness Checks."""
+
+import logging
+
+from . import errors
+from .installer import is_installed
+from .multipass import Multipass
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_multipass_is_ready(*, multipass: Multipass = Multipass()) -> None:
```

Comment:
I still wonder why we have a default here and if you could just require it. I would argue that being the of the incorrect version is not not being ready and maybe a name change to `ensure_multipass_environment` or similar would be a bit more appropriate.

Reply:
I agree the name is a bit odd.  I'd like to wrap up this functionality into a generalized `provider.initialize()` API, but I have been delaying that until things have firmed up on the Charmcraft side and we expand to other craft tools.

The default value is just consistent with the rest of the APIs.  Worth reconsidering that too at some point.